### PR TITLE
x86: never set ARCH_HAS_USERSPACE

### PR DESCRIPTION
**Pull request by tyrantlizard77**
The Meltdown bug is of significant concern and we shall disable User
mode on x86 until this issue is addressed.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

### CODE REVIEW

**Review by lpereira — APPROVED** [review]

**Review by nashif — APPROVED** [review]

File: tests/kernel/userbufer_validate/testcase.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 tests:
      - test:
-        tags: kernel
-        platform_whitelist: qemu_x86 arduino_101
```
Comment (andrewboie) [inline tests/kernel/userbufer_validate/testcase.yaml]: this is not quite right, this test is very x86 specific

**Comment by nashif** [comment]
CI

**Comment by nashif** [comment]
CI now building with sphinx 1.6.5, is this still needed?

**Comment by stephensmalley** [comment]
This seems like an overreaction.  I'd still rather have userspace support even without a fix for meltdown; it still provides benefits.

**Comment by nashif** [comment]
> This seems like an overreaction. I'd still rather have userspace support even without a fix for meltdown; it still provides benefits.

Note this change applies to 1.10, not master.

**Pull request closed without merging**